packages/react-dev-utils/WebpackDevServerUtils.js

```diff
@@ -152,9 +159,32 @@ function createCompiler(webpack, config, appName, urls, useYarn) {
     // them in a readable focused way.
     // We only construct the warnings and errors for speed:
     // https://github.com/facebook/create-react-app/issues/4492#issuecomment-421959548
-    const messages = formatWebpackMessages(
-      stats.toJson({ all: false, warnings: true, errors: true })
-    );
+    const statsData = stats.toJson({ all: false, warnings: true, errors: true });
+
+    if (useTypeScript && statsData.errors.length === 0) {
+      process.stdout.write(
+        chalk.yellow('Files successfully emitted, waiting for typecheck results...')
```

Comment:
Is the thinking here that the dev server would already be in the process of refreshing the browser with the updated source at this point, but that the type checking could take a while longer?

Reply:
Yes, it could flicker in a first build as TS is probably faster than Babel. But in the following rebuilds, Babel is almost always faster than TS. So, the idea is to emit files as quick as possible and remind the user that type-check is still in progress. Webpack actually emit files just before it call *done* hook.